flocker/node/agents/blockdevice.py

```diff
@@ -1673,10 +1673,9 @@ def is_existing_block_device(dataset_id, path):
                 applications=None,
                 used_ports=None,
             ),
+            NonManifestDatasets(datasets=nonmanifest),
         )
 
-        if nonmanifest:
-            state += (NonManifestDatasets(datasets=nonmanifest),)
```

Comment:
Okay.  Can't think of a good reason it would be important to not report these if the value is empty.